crates/ra_hir/src/ty/infer.rs

```diff
@@ -489,36 +490,38 @@ impl<'a, D: HirDatabase> InferenceContext<'a, D> {
             // Attempt to find an impl_item for the type which has a name matching
             // the current segment
             log::debug!("looking for path segment: {:?}", segment);
-            let item: crate::ModuleDef = ty.iterate_impl_items(self.db, |item| {
-                let matching_def: Option<crate::ModuleDef> = match item {
-                    crate::ImplItem::Method(func) => {
-                        let sig = func.signature(self.db);
-                        if segment.name == *sig.name() {
-                            Some(func.into())
-                        } else {
-                            None
+            let item: crate::ModuleDef = krate.and_then(|k| {
```

Comment:
I think instead of indenting everything here further, you can just use `?` on the krate. I don't think it's actually possible that we get to this point without a crate graph.

Reply:
👌 great, I wasn't sure about it :)